File: Java Solution/SelectionSort.java
public class SelectionSort {
    private int[] numbers;

    public SelectionSort(int[] numbers) {
        this.numbers = numbers;
    }

    public void sort() {
        int n = numbers.length;
        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < n; j++) {
                if (numbers[j] < numbers[minIndex]) {
                    minIndex = j;
                }
            }
            int temp = numbers[i];
            numbers[i] = numbers[minIndex];
            numbers[minIndex] = temp;
            printArray();
        }
        printArray();
    }

    public void printArray() {
        for (int num : numbers) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] data = {64, 25, 12, 22, 11};
        SelectionSort sorter = new SelectionSort(data);
        sorter.sort();
    }
}

File: C++ Solution/SelectionSort.cpp
#include <iostream>
#include <vector>
using namespace std;

class SelectionSort {
public:
    vector<int> numbers;

    SelectionSort(vector<int> arr) {
        numbers = arr;
    }

    void sort() {
        int n = numbers.size();
        for (int i = 0; i < n - 1; i++) {
            int min_index = i;
            for (int j = i + 1; j < n; j++) {
                if (numbers[j] < numbers[min_index]) {
                    min_index = j;
                }
            }
            swap(numbers[i], numbers[min_index]);
            printArray();
        }
        printArray();
    }

    void printArray() {
        for (int num : numbers) {
            cout << num << " ";
        }
        cout << endl;
    }
};

int main() {
    vector<int> data = {64, 25, 12, 22, 11};
    SelectionSort sorter(data);
    sorter.sort();
    return 0;
}
